### ISSUE TITLE
sddm 0.21.0 not start custom desktop session, it works on sddm 0.19.0

### ISSUE THREAD

**Issue by katnatek** [sddm/sddm#1982]
I have a custom desktop session that makes my user start lxqt session and the other user start plasma x11 session,.
That was working in sddm 0.19.0 but not work in sddm 0.21.0

A packager in mageia points that is trigerred by trailing spaces https://bugs.mageia.org/show_bug.cgi?id=33493#c4
This is the desktop file installed in /usr/share/xsessions/
[smart.desktop.txt](https://github.com/user-attachments/files/16806590/smart.desktop.txt)

And this is the script with  execution rights in /usr/sbin
[startDE.sh.txt](https://github.com/user-attachments/files/16806884/startDE.sh.txt)

All I can see in the desktop list to select is an extra "empty" when in 0.19.0 was the custom name and a message in the journal

Failed to find command for session "/usr/share/xsessions/smart.desktop"

**Comment by Vogtinator** [comment]
According to (my interpretation of) https://specifications.freedesktop.org/desktop-entry-spec/latest/basic-format.html#group-header, it's not allowed to have leading or trailing spaces around group headers like `[Desktop Entry]`, so the .desktop file is invalid:

```
00000000  5b 44 65 73 6b 74 6f 70  20 45 6e 74 72 79 5d 20  |[Desktop Entry] |
00000010  0a 54 79 70 65 3d 41 70  70 6c 69 63 61 74 69 6f  |.Type=Applicatio|
```

We could add a warning in SDDM and try to handle this gracefully, but the .desktop file needs to be fixed.

**Comment by katnatek** [comment]
> According to (my interpretation of) https://specifications.freedesktop.org/desktop-entry-spec/latest/basic-format.html#group-header, it's not allowed to have leading or trailing spaces around group headers like `[Desktop Entry]`, so the .desktop file is invalid:
> 
> We could add a warning in SDDM and try to handle this gracefully, but the .desktop file needs to be fixed.

If you can do it would be good, thank you for the information and the attention, I remove the extra spaces and now it works.
This is not the first time I have to fix a desktop file due to more strict use of specifications, I let the decision in your hands

**Comment by Vogtinator** [comment]
I played around a bit with `kreadconfig6` and it ignores trailing space everywhere - even for values (which is IMO a mistake in the spec, it's not clear enough and makes it ambiguous, so not implementing that here unless necessary).

With https://github.com/sddm/sddm/pull/1983, the "smart.desktop" is now accepted with a warning and should work.

**Comment by katnatek** [comment]
I apply the changes, the custom session is listed, but the action is not performed (the desktop session not start)  with the defective desktop file and the message Failed to find command for session "/usr/share/xsessions/smart.desktop" in journal is added

Can you confirm that is the expected behavior?

**Comment by Vogtinator** [comment]
No, not at all - it should work. I don't see why it wouldn't, the error implies the `Exec` value is empty. Is your .desktop file really the same as you uploaded?

**Comment by katnatek** [comment]
> No, not at all - it should work. I don't see why it wouldn't, the error implies the `Exec` value is empty. Is your .desktop file really the same as you uploaded?

Yes it Is, I have the file in other location, and restore the original /usr/share/xsessions/smart.desktop by cp -f backup /usr/share/xsessions/smart.desktop , then test with untouched sddm again, update with custom build of sddm with the changes in https://github.com/sddm/sddm/pull/1983 , I will test again rebooting the system instead of just logout and let you know the result

**Comment by katnatek** [comment]
OK a full reboot did the trick, sorry for the noise, you can close this, I already send to mageia a patch  with the suggested changes.
